rescript/ncbi.py

```diff
@@ -0,0 +1,141 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import requests
+from xmltodict import parse
+from pandas import DataFrame
+from q2_types.feature_data import DNAIterator
+from skbio import DNA
+from qiime2 import Metadata
+
+
+def get_ncbi_data(query: str = None, accession_ids: Metadata = None) -> (
+        DNAIterator, DataFrame):
+    if query is None and accession_ids is None:
+        raise ValueError('Query or accession_ids must be supplied')
+
+    if query:
+        seqs, taxids = get_nuc_for_query(query)
+
+    if accession_ids:
+        accs = accession_ids.get_ids()
+        if query and seqs:
+            accs = accs - seqs.keys()
+            if accs:
+                acc_seqs, acc_taxids = get_nuc_for_accs(accs)
+                seqs.update(acc_seqs)
+                taxids.update(acc_taxids)
+        else:
+            seqs, taxids = get_nuc_for_accs(accs)
+
+    taxa = get_taxonomies(taxids)
+
+    seqs = DNAIterator(DNA(v, metadata={'id': k}) for k, v in seqs.items())
```

Comment:
why create a DNAIterator? You could create an empty DNAFASTAFormat object, then just parse and save seqs to that as they are generated

Reply:
I did it that way because that's the way I've always returned something to be saved as a FeatureData[Sequence]. Is creating a DNAFASTAFormat object preferable?

Reply:
Yes, I think `DNAFASTAFormat` is better here.

Reply:
actually... what is `seqs` here? looks like it is a dict? in which case creating an iterator to save makes sense.

Reply:
Thanks!